Code/Framework/AzToolsFramework/AzToolsFramework/Archive/ArchiveComponent.cpp

```diff
@@ -141,320 +120,474 @@ namespace AzToolsFramework
-    {
-        AZStd::string commandLineArgs = Platform::GetExtractArchiveCommand(archivePath, destinationPath, true);
+            bool success = true;
+            AZStd::vector<char> fileBuffer;
+            const AZ::IO::Path workingPath{ dirToArchive };
 
-        if (commandLineArgs.empty())
-        {
-            // The platform-specific implementation has already thrown its own error, no need to throw another one
-            return;
-        }
+            for (const auto& fileName : foundFiles.GetValue())
+            {
+                bool thisSuccess = false;
 
-        LaunchZipExe(m_unzipExePath, commandLineArgs, respCallback, taskHandle);
-    }
+                AZ::IO::PathView relativePath = AZ::IO::PathView{ fileName }.LexicallyRelative(workingPath);
 
-    void ArchiveComponent::ExtractArchiveWithoutRoot(const AZStd::string& archivePath, const AZStd::string& destinationPath, AZ::Uuid taskHandle, const ArchiveResponseOutputCallback& respCallback)
-    {
-        AZStd::string commandLineArgs = Platform::GetExtractArchiveCommand(archivePath, destinationPath, false);
+                AZStd::string fullPath = (workingPath / relativePath).c_str();
+                if (ArchiveUtils::ReadFile(fullPath, AZ::IO::OpenMode::ModeRead, fileBuffer))
+                {
+                    int result = archive->UpdateFile(
+                        relativePath.Native(), fileBuffer.data(), fileBuffer.size(), s_compressionMethod,
+                        s_compressionLevel, s_compressionCodec);
+
+                    thisSuccess = (result == AZ::IO::ZipDir::ZD_ERROR_SUCCESS);
+                    AZ_Error(
+                        s_traceName, thisSuccess, "Error %d encountered while adding '%s' to archive '%.*s'", result, fileName.c_str(),
+                        AZ_STRING_ARG(archive->GetFullPath().Native()));
+                }
+                else
+                {
+                    AZ_Error(
+                        s_traceName, false, "Error encountered while reading '%s' to add to archive '%.*s'", fileName.c_str(),
+                        AZ_STRING_ARG(archive->GetFullPath().Native()));
+                }
 
-        if (commandLineArgs.empty())
-        {
-            // The platform-specific implementation has already thrown its own error, no need to throw another one
-            return;
-        }
+                success = (success && thisSuccess);
+            }
+
+            archive.reset();
+            p.set_value(success);
+        };
 
-        LaunchZipExe(m_unzipExePath, commandLineArgs, respCallback, taskHandle);
+        // Async task...
+        std::promise<bool> p;
+        std::future<bool> f = p.get_future();
+
+        AZStd::thread_desc threadDesc;
+        threadDesc.m_name = "Archive Task (Create)";
+        m_threads.emplace_back(threadDesc, FnCreateArchive, AZStd::move(p));
+        return AZStd::move(f);
```

Comment:
The move isn't needed and inhibits guaranteed copy elision in C++17: https://stackoverflow.com/questions/19267408/why-does-stdmove-prevent-rvo

Reply:
Will update the returns.